### ISSUE TITLE
NeighborSampler Segfault when data on GPU

### ISSUE THREAD

**Issue by arthurfeeney** [rusty1s/pytorch_geometric#986]
## 🐛 Bug

When using NeighborSampler with data that is on the GPU, there is a segfault. I believe this is caused by the neighbor_sampler function. In particular, self.cumdeg being on the GPU seems to be the culprit. 

## To Reproduce

1. generate some data and use .to(cuda_device) on it
2. build a NeighborSampler with it (I don't think any of the other parameters affect the error)
3. call the sampler, and it will segfault. 

~~~~
import torch                                                                   
from torch_geometric.data import Data, NeighborSampler                         
from torch_geometric.utils import erdos_renyi_graph                            
                                                                              
device = torch.device('cuda')                                                  
num_nodes = 10                                                                 
data = Data(edge_index=erdos_renyi_graph(num_nodes, 0.1)).to(device)           
data.num_nodes = num_nodes                                                     
                                                                           
loader = NeighborSampler(data, size=[4,0.5], num_hops=2, batch_size=2,         
                                            shuffle=True, bipartite=True, add_self_loops=False)   
                                                                              
# segfault when data is on GPU, fine if on CPU                                 
for sample in loader():                                                        
    print('hi :)') 
~~~~

Here is the error this sample gave. 
~~~~
/cm/local/apps/slurm/var/spool/job6215756/slurm_script: line 14: 17674 Segmentation fault      python geo_err.py
~~~~  

## Expected behavior
I was expecting it work whether or not the data is on the GPU. (I was able to get it to work by copy-pasting NeighborSampler and adding in .to(device) and .cpu() in the needed places, but probably not a great solution). 
If it is weird to have data on the GPU, I think there should at least be an assertion or something that checks which device the data is on. 

## Environment

* OS: CentOS 7
* Python version: 3.8.1
* PyTorch version: 1.4.0
* CUDA/cuDNN version: 10.1
* GCC version: 5.4

**Comment by rusty1s** [comment]
As you correctly noticed, the current `NeighborSampler` only has a CPU implementation. That makes sense since you want to use it if the current `data` cannot find onto the GPU. I will add an assertion to prevent this usage :)

**Comment by SKD621** [comment]
@rusty1s It seems that the NeighborSampler is still unavailable on GPU？ (my torch geometric version is 1.4.3)
